F1/F1/CoreData/CoreDataManager.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  CoreDataManager.swift
+//  F1
+//
+//  Created by Kaitlyn Sookdhew on 2024/05/15.
+//
+import CoreData
+import Foundation
+import UIKit
+
+class CoreDataManager {
+    static let shared = CoreDataManager()
+    let appDelegate: AppDelegate
+    let context: NSManagedObjectContext
+
+    init() {
+        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else {
+            fatalError("Unable to access AppDelegate")
+        }
+
+        self.appDelegate = appDelegate
+        context = appDelegate.persistentContainer.viewContext
+    }
+
+    func saveRaceInfo(_ racing: Racing) {
+
+        let fetchRequest: NSFetchRequest<NSFetchRequestResult> = CoreDataRaceInfo.fetchRequest()
+        let deleteRequest = NSBatchDeleteRequest(fetchRequest: fetchRequest)
+
+        do {
```

Comment:
like this do try save function can just be in a helper function called saveCoreData() and be called everywhere you sve instead of reusing the whole do try catch block